components/SearchBox.vue

```diff
@@ -0,0 +1,179 @@
+<template>
+  <div
+    class="overflow-x-auto border border-silver-200"
+  >
+    <table class="min-w-full bg-white table-auto">
+      <thead class="sticky top-0 z-1">
+        <tr class="text-left headers text-white h-14 text-lg">
+          <th
+            v-for="column in filteredColumns"
+            :key="column.key"
+            class="py-2 px-4 border-b border-gray-200"
+          >
+            {{ column.key }}
+          </th>
+        </tr>
+
+        <!-- Filters Row -->
+        <tr>
+          <th
+            v-for="column in filteredColumns"
+            :key="column.key"
+            class="text-lg py-2 px-4 border-y border-gray-200 bcgovgold"
+          >
+            <!-- Render text input for specified columns -->
+            <input
+              v-if="column.name !== 'NatureOfBusiness' && column.name !== 'LastComment'&&
+                ['LastModifiedBy','NameRequestNumber', 'Names', 'ApplicantFirstName', 'ApplicantLastName']
+                  .includes(column.name)"
+              v-model="columnFilters[column.name]"
+              type="text"
+              :placeholder="column.key"
+              class="border rounded px-2 py-1 w-full"
+              @keyup.enter="handleFilterChange"
```

Comment:
Up to you but I prefer to see the event handler as a function here, eg, `handleFilterChange($event)` -- it looks better / easier to understand in my code editor.

Reply:
Thanks, much appreciated! (Easy reviews = good code)